File: app_pasos/lib/models/workout.dart
class WorkoutExercise {
  final String exerciseId;
  final String exerciseName;
  final int setsCompleted;
  final String repsCompleted;

  WorkoutExercise({
    required this.exerciseId,
    this.exerciseName = '',
    this.setsCompleted = 0,
    this.repsCompleted = '',
  });

  factory WorkoutExercise.fromJson(Map<String, dynamic> json) {
    return WorkoutExercise(
      exerciseId: json['exercise'] ?? '',
      exerciseName: json['exerciseName'] ?? '',
      setsCompleted: json['setsCompleted'] ?? 0,
      repsCompleted: json['repsCompleted'] ?? '',
    );
  }
}

class Workout {
  final String id;
  final String routineName;
  final DateTime date;
  final int duration;
  final List<WorkoutExercise> exercises;

  Workout({
    required this.id,
    this.routineName = '',
    required this.date,
    this.duration = 0,
    this.exercises = const [],
  });

  factory Workout.fromJson(Map<String, dynamic> json) {
    return Workout(
      id: json['_id'] ?? json['id'] ?? '',
      routineName: json['routineName'] ?? '',
      date: json['date'] != null ? DateTime.parse(json['date']) : DateTime.now(),
      duration: json['duration'] ?? 0,
      exercises: (json['exercises'] as List?)
              ?.map((e) => WorkoutExercise.fromJson(e))
              .toList() ??
          [],
    );
  }
}

File: backend/models/Workout.js
const mongoose = require('mongoose');

const workoutExerciseSchema = new mongoose.Schema({
  exercise: {
    type: String,
    required: true
  },
  exerciseName: String,
  setsCompleted: {
    type: Number,
    default: 0
  },
  repsCompleted: {
    type: String,
    default: ''
  },
  weightKg: {
    type: Number,
    default: 0
  }
});

const workoutSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  routine: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Routine',
    default: null
  },
  routineName: {
    type: String,
    default: ''
  },
  date: {
    type: Date,
    required: true
  },
  duration: {
    type: Number,
    default: 0
  },
  exercises: [workoutExerciseSchema]
}, {
  timestamps: true
});

workoutSchema.index({ user: 1, date: -1 });

module.exports = mongoose.model('Workout', workoutSchema);